### ISSUE TITLE
Play,Pause and Stop didn't Work for Slave Speakers

### ISSUE THREAD

**Issue by Missenberger79** [hklages/node-red-contrib-sonos-plus#31]
Hello,

if I send Pause  to a Player who is a Slave I get 

`pause - Request failed with status code 500 :: Details: {"stack":"Error: Request failed with status code 500\n    at createError (/opt/iobroker/iobroker-data/node-red/node_modules/sonos/node_modules/axios/lib/core/createError.js:16:15)\n    at settle (/opt/iobroker/iobroker-data/node-red/node_modules/sonos/node_modules/axios/lib/core/settle.js:17:12)\n    at IncomingMessage.handleStreamEnd (/opt/iobroker/iobroker-data/node-red/node_modules/sonos/node_modules/axios/lib/adapters/http.js:236:11)\n    at IncomingMessage.emit (events.js:203:15)\n    at endReadableNT (_stream_readable.js:1143:12)\n    at process._tickCallback (internal/process/next_tick.js:63:19)","message":"Request failed with status code 500","config":{}}`

On a Play-Value  I get the message OK:play , but nothing happens. 

On a stop-Value  I get the message OK:stop , but only the slave stopped.

**Comment by hklages** [comment]
Hello, 

YES that's true. Many of the commands have unpredictable results when being send to a "slave". 

My recommendation: Don't send any commands to slaves except volume adjustment, mute, unmute (these command should work also at the slave) - and leave_group. There is now the possibility with "get_groups" in Get Status node to check whether that player is master, slave, independent. 

I consider to revise the overall "group" behavior. Any suggestions are welcome. 
One option could be: throw an error message to state that this mode is not supported for a slave

**Comment by Missenberger79** [comment]
The problem is that I have installed KNX switches in the house. Here I have linked a KNX NODE Play Pause or Stop with your node. 

Because I don't always have the bathroom as master, for example, the error occurs again. But I don't always want to use the app to switch off my group e.g. in the morning before work.

In another Sonos NODE, I think I read that it is possible to make a speaker the master of a group.  Maybe it would be possible like you did with the suppressWarnings, but the other way around. So sending a command before start/stop makes me the master.

**Comment by hklages** [comment]
I just published 2.1.0. There an error message is thrown, in case you send a command to a client in a group.

But in parallel your recent comment shows up. I guess 2.1.0 will not solve your issue. 

Can you write down a workflow that I better understand - in English or German please. 

By the way: issues 21, 22 are open and i haven received any feedback from you.

**Comment by Rolf-M** [comment]
I would say, the play/pause/precious/next... of a slave speaker have to be redirected to the master, if they are sent to a slave. So, no matter what slave you will address all the execution is done on the master. This is pretty close to how the app works, as you only have a "group virtual speaker" once you group some speakers.

**Comment by hklages** [comment]
... will come soon. 

There is a already the first (test) command `play_notif` ( a revised version of `play_notification`) and there you can specify msg.useCoordinator (default is true). If you use msg.volume in combination with msg.sameValue (default is also true) it will play the notification on all members of that group.

**Comment by hklages** [comment]
There is now a new node - Universal. Here all commands (play, toggle.playback, stop, ...) are GROUP commands. You can also use msg.playerName=kitchen  - it will overwrite the player in the configNode - but keep in mind, this will also go to the GROUP. 

For your issue: would it help to have a player.play? In case the player is a joiner (that means not the coordinator of the group) it leaves the group and plays the song? 

For notification, I implemented that: jointer.play.notification

**Comment by hklages** [comment]
timed out - please open a new issue if it still exist.
